lib/auth.js

```diff
@@ -561,241 +341,6 @@ api.setEmail = function(actor, id, email, callback) {
-        return api.verifyPassword(identity, callback);
-      }
-      callback(null, null);
-    }],
-    checkPasscode: ['checkPermission', function(callback) {
-      if('sysPasscode' in identity) {
-        return api.verifyPasscode(identity, callback);
-      }
-      callback(null, null);
-    }],
-    hashPassword: ['checkPassword', 'checkPasscode', function(
-      callback, results) {
-      if(results.checkPassword === false) {
-        return callback(new BedrockError(
-          'Could not update identity password; invalid password.',
-          'InvalidPassword'));
-      }
-      if(results.checkPasscode === false) {
-        return callback(new BedrockError(
-          'Could not update identity passcode; invalid passcode.',
-          'InvalidPasscode'));
-      }
-      if('sysPasswordNew' in identity) {
-        return api.createPasswordHash(identity.sysPasswordNew, callback);
-      }
-      callback();
-    }],
-    generatePasscode: ['hashPassword', function(callback, results) {
-      if(results.hashPassword) {
-        changes.sysPassword = results.hashPassword;
-      }
-      var passcode = identity.sysPasscode = _generatePasscode();
-      api.createPasswordHash(passcode, callback);
-    }],
-    update: ['generatePasscode', function(callback, results) {
-      changes.sysPasscode = results.generatePasscode;
-      database.collections.identity.update(
-        {id: database.hash(identity.id)},
-        {$set: database.buildUpdate(changes, 'identity')},
-        database.writeOptions,
-        function(err, result) {
-          if(err) {
-            return callback(err);
-          }
-          if(result.result.n === 0) {
-            return callback(new BedrockError(
-              'Could not set Identity password. Identity not found.',
-              'NotFound'));
-          }
-          callback();
-        });
-    }]
-  }, function(err) {
-    callback(err, changes);
-  });
-};
-
-/**
- * Verifies the Identity's password against the stored password.
- *
- * @param identity the Identity with the password to verify.
- * @param callback(err, verified) called once the operation completes.
- */
-api.verifyPassword = function(identity, callback) {
-  _verifyPasswordHash(identity, 'password', callback);
-};
-
-/**
- * Verifies the Identity's passcode against the stored passcode.
- *
- * @param identity the Identity with the passcode to verify.
- * @param callback(err, verified) called once the operation completes.
- */
-api.verifyPasscode = function(identity, callback) {
-  _verifyPasswordHash(identity, 'passcode', callback);
-};
-
-/**
- * Verifies the Identity's passcode against the stored passcode and sets
- * the Identity's email address as verified upon success.
- *
- * @param actor the Identity performing the action.
- * @param identity the Identity with the passcode to verify.
- * @param callback(err, verified) called once the operation completes.
- */
-api.verifyEmail = function(actor, identity, callback) {
```

Comment:
This is now in `bedrock-authn-password` -- let's mark it deprecated and call into that module and make this a simple alias.